src/common/actions/create-snap.js

```diff
@@ -81,30 +81,45 @@ export function createSnap(repositoryUrl, location) { // location for tests
         .then(checkStatus)
         .then(response => {
           return response.json().then(result => {
-            if (result.status !== 'success' ||
-                result.payload.code !== 'snap-created') {
-              throw getError(response, result);
+            if (result.status !== 'success' || result.payload.code !== 'snap-created') {
+              return Promise.reject(getError(response, result));
```

Comment:
`createSnap` is now called by `createSnaps`, so instead of handling the success and failure scenarios here they are now handled by the `createSnaps` promise.